src/tree/updater_gpu_hist.cu

```diff
@@ -38,6 +38,7 @@ struct GPUHistMakerTrainParam
   bool single_precision_histogram;
   // number of rows in a single GPU batch
   int gpu_batch_nrows;
+  bool debug_synchronize;
```

Comment:
Not to cause you more troubles.  But can we use existing `verbosity:debug` flag?  We have `debug_verbose` before and somehow people take it as part of the interface.  Even though I deprecated by introducing `verbosity` parameter, I'm still seeing its use in issues.  Be careful when adding flags like these...

Reply:
You can use `ConsoleLogger::ShouldLog`, see its use in `timer.h::Monitor` for an example.

Reply:
I don't want this to be a part of verbose due to the hidden performance penalty. This flag may not be a permanent fixture.